plugin-api/src/main/java/org/sonar/api/web/ServletFilter.java

```diff
@@ -19,197 +19,23 @@
  */
 package org.sonar.api.web;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Set;
-import java.util.function.Predicate;
-import java.util.stream.Collectors;
-import javax.servlet.Filter;
 import org.sonar.api.ExtensionPoint;
 import org.sonar.api.server.ServerSide;
 
-import static java.util.Arrays.asList;
-import static java.util.Collections.unmodifiableList;
-import static org.apache.commons.lang.StringUtils.substringBeforeLast;
-import static org.sonar.api.utils.Preconditions.checkArgument;
-
 /**
+ * {@code @deprecated} since 9.16. Use {@link org.sonar.api.web.HttpFilter} instead.
+ *
  * @since 3.1
  */
 @ServerSide
 @ExtensionPoint
-public abstract class ServletFilter implements Filter {
+@Deprecated(forRemoval = true, since = "9.16")
+public abstract class ServletFilter implements javax.servlet.Filter {
 
   /**
    * Override to change URL. Default is /*
    */
   public UrlPattern doGetPattern() {
     return UrlPattern.builder().build();
   }
-
-  public static final class UrlPattern {
```

Comment:
I don't think we can drop this class, as it will break compatibility with current plugins - import change